git-lfs.go

```diff
@@ -1,11 +1,37 @@
 package main
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+
 	"github.com/github/git-lfs/commands"
 	"github.com/github/git-lfs/lfs"
 )
 
 func main() {
+	c := make(chan os.Signal)
+	signal.Notify(c)
```

Comment:
I think you should actually specify the types i.e. `signal.Notify(c, os.Interrupt, os.Kill)` to avoid overrunning the channel buffer, the docs say: 
> Package signal will not block sending to c: the caller must ensure that c has sufficient buffer space to keep up with the expected signal rate. For a channel used for notification of just one signal value, a buffer of size 1 is sufficient.
Kill/Interrupt are OK with a single buffer value because they're only sent once & then we terminate, but other non-terminating signals can get sent more frequently. Sounds like they might either overwrite or overrun a single-entry channel in that case.